scheduler/replica_scheduler.go

```diff
@@ -67,6 +63,9 @@ func (rcs *ReplicaScheduler) ScheduleReplica(replica *longhorn.Replica) (*longho
 	// TODO just set default directory for now
 	replica.Spec.DataPath = longhornDirectory + "/replicas/" + replica.Spec.VolumeName + "-" + util.RandomID()
 
+	// set current replica back to replica map
```

Comment:
Should the caller take care of it?

Reply:
I just deleted current replica from the map at the beginning of scheduler for scheduling, I think I need to add it back after scheduling.